sw/device/boot_rom2/ext_format.c

```diff
@@ -0,0 +1,130 @@
+// Cn:opyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/boot_rom2/ext_format.h"
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/runtime/check.h"
+
+static const uint32_t kExtMagic = 0x5245545f;
+
+/**
+ * Convenience struct for pulling out the fixed part of the executable header.
+ */
+typedef struct fixed_header {
+  uint32_t magic, version, payload_len, entrypoint;
```

Comment:
There are other things that will be added here later, but a key_id or hash of the pub_key used to verify the signature may be useful to integrate since rom will have multiple keys (potentially with different purposes, e.g. TEST versus DEV versus PROD ).

Reply:
This is a first pass of the format. I'm expecting that kind of stuff (and whether it should go in the permanent header or in versioned metadata) to fall out of a more thorough security review.

Reply:
ack. Re. the versioned metadata comment, the key identifier should go with the signature.